### ISSUE TITLE
Hard coupling between com_users and plg_user_joomla

### ISSUE THREAD

**Issue by mbabker** [joomla/joomla-cms#20249]
Within [`UsersModelUser::getForm()`](https://github.com/joomla/joomla-cms/blob/49e681fc66ecdd76388fa9e9cb3a92986c075e69/administrator/components/com_users/models/user.php#L131-L138) there is logic for the send mail to users plugin capability that is hardcoded to only support that plugin.  Either this capability should be fully moved to the plugin (meaning the form alterations should happen via plugin event) or be fully moved to the component (meaning the mail sending code exists in the component).

**Comment by carlitorweb** [comment]
I on it, in my opinion, this need be inside the component. But this will create a b/c compatibility right?

**Comment by mbabker** [comment]
The safest plan would be to move the logic from the component to the plugin and use the plugin events to modify the form (that wouldn't be a B/C break by any sane definition) instead of moving the feature support from the plugin to the component (which probably gets into B/C concerns because it means configuration checking needs to account for both the existing plugin config and a new component config value).

**Comment by carlitorweb** [comment]
I hate every time something needs the safest plan (joke). 
Well, then if make the plugin modify the form for set as required the password field, for avoid B/C breaks, then, will be good make another PR to J4, moving all the logic to the component (this, because I think is the right place for this code). Is a good plan?

**Comment by mbabker** [comment]
I wouldn't put it in the component.  With a plugin driven approach it's possible to use a different email (once the hard coupling is in place), given the lack of events in our email sending process the next best thing is leaving this in the plugin and letting someone write another plugin with a different email (or logic around sending it) if they want to customize it any.

**Comment by ghost** [comment]
closed as having Pull Request #20275.
